node/src/Commands.ts

```diff
@@ -803,42 +803,175 @@ export function createZscore(
     return createCommand(RequestType.ZScore, [key, member]);
 }
 
-export type ScoreLimit =
+export type ScoreLimit<T> =
+    /**
+     * Positive infinity bound for sorted set.
+     */
     | `positiveInfinity`
+    /**
+     * Negative infinity bound for sorted set.
+     */
     | `negativeInfinity`
+    /**
+     *  Represents a specific numeric score boundary in a sorted set.
+     */
     | {
-          bound: number;
+          /**
+           * The score value.
+           */
+          value: T;
+          /**
+           * Whether the score value is inclusive. Defaults to True.
+           */
           isInclusive?: boolean;
       };
 
-function getScoreLimitArg(score: ScoreLimit): string {
+/**
+ * Represents a range by index (rank) in a sorted set.
+ * The `start` and `stop` arguments represent zero-based indexes.
+ */
+export type RangeByIndex = {
+    /**
+     *  The start index of the range.
+     */
+    start: number;
+    /**
+     * The stop index of the range.
+     */
+    stop: number;
+};
+
+/**
+ * Represents a range by score or a range by lex in a sorted set.
+ * The `start` and `stop` arguments represent score boundaries.
+ */
+type Range<T> = {
```

Comment:
Please use a more specific name - if this range can be used in sorted and non-sorted sets, maybe something like SetRange, otherwise maybe SortedSetRange, or something similar.. see how it can be both generic for other usage and not too generic (Range) so it wouldn't be confusing